librz/core/disasm.c

```diff
@@ -6815,3 +6815,55 @@ RZ_API RZ_OWN char *rz_core_disasm_instruction(RzCore *core, ut64 addr, ut64 rel
 	}
 	return buf_asm;
 }
+
+RZ_API void rz_core_disasm_op_free(RzCoreDisasmOp *x) {
+	if (!x) {
+		return;
+	}
+	free(x->assembly);
+	free(x->assembly_colored);
+	free(x->hex);
+	free(x);
+}
+
+/**
+ * \brief Disassemble all possible opcodes (byte per byte) at \p addr
+ */
+RZ_API RZ_OWN RzPVector /*<RzCoreDisassemblyOp *>*/ *rz_core_disasm_all_possible_opcodes(RZ_NONNULL RzCore *core, RZ_NONNULL ut8 *buffer, ut64 addr, ut64 n_bytes) {
```

Comment:
@Rot127 do you think this function can be rewritten by using your new code for colors? It feels to me the `RzCoreDisassemblyOp` structure might be a bit redundant here.

Reply:
I don't see how this could be done with the new TokenStrings. Since  `RzCoreDisassemblyOp` holds more information then the `RzAsmTokenString`, `rz_core_disasm_all_possible_opcodes` had to return the complete `AsmOp` if `RzCoreDisassemblyOp` would be removed (so the offsets and size can be provided to the handlers).

Personally I don't mind the additional struct and I like that the handlers are not too bloated with `RzAsmOp` stuff. Maybe a doxygen could be added to `RzCoreDisassemblyOp` so it isn't used at places where `RzAsmOp` would be sufficient?

Reply:
Yes, we've had enough in terms of repeatability, like `RzAsmOp` `RzAnalysisOp` `RzCoreAsmHit` `RzAnalysisBytes`